lib/minitest/reporters/base_reporter.rb

```diff
@@ -86,8 +86,12 @@ def print(*args)
       end
 
       def print_info(e, name=true)
-        print "#{e.exception.class.to_s}: " if name
-        e.message.each_line { |line| print_with_info_padding(line) }
+        if name
+          print "#{e.exception.class.to_s}: "
+          e.exception.message.each_line { |line| print_with_info_padding(line) }
+        else
+          e.message.each_line { |line| print_with_info_padding(line) }
```

Comment:
`name` is true when `e` is a `UnexpectedError`. `name` is false when `e` is an `Assertion`. You have to look at the definitions of those two classes' `message` method to know why this is the case